internal/extsvc/npm/npm_test.go

```diff
@@ -103,3 +163,14 @@ func TestFetchSources(t *testing.T) {
 		"package/test/index.js",
 	})
 }
+
+func TestNoPanicOnNonexistentRegistry(t *testing.T) {
+	ctx := context.Background()
+	client, stop := newTestHTTPClient(t, true)
+	defer stop()
+	client.registryURL = "http://not-an-npm-registry.sourcegraph.com"
+	dep, err := reposource.ParseNPMDependency("left-pad@1.3.0")
+	require.Nil(t, err)
+	_, err = client.DoesDependencyExist(ctx, *dep)
+	require.NotNil(t, err)
```

Comment:
Give it some room to breath 😄 

```suggestion
	ctx := context.Background()
	
	client, stop := newTestHTTPClient(t, true)
	defer stop()
	
	client.registryURL = "http://not-an-npm-registry.sourcegraph.com"
	
	dep, err := reposource.ParseNPMDependency("left-pad@1.3.0")
	require.Nil(t, err)
	
	_, err = client.DoesDependencyExist(ctx, *dep)
	require.NotNil(t, err)
```